openmc/deplete/atom_number.py

```diff
@@ -47,8 +45,8 @@ class AtomNumber:
 
     """
     def __init__(self, local_mats, nuclides, volume, n_nuc_burn):
-        self.index_mat = OrderedDict((mat, i) for i, mat in enumerate(local_mats))
-        self.index_nuc = OrderedDict((nuc, i) for i, nuc in enumerate(nuclides))
+        self.index_mat = {(mat, i) for i, mat in enumerate(local_mats)}
+        self.index_nuc = {(nuc, i) for i, nuc in enumerate(nuclides)}
```

Comment:
I think these lines are causing the ci fails

```
def _get_mat_index(self, mat):
        """Helper method for getting material index"""
        if isinstance(mat, Material):
            mat = str(mat.id)
>       return self.index_mat[mat] if isinstance(mat, str) else mat
E       TypeError: 'set' object is not subscriptable
```

Reply:
solved with [0be3d05](https://github.com/openmc-dev/openmc/pull/2641/commits/0be3d059f4a1c507331f808a653474647f348922)